src/Test.UnitTests.Sarif.Driver/Sdk/AnalyzeCommandBaseTests.cs

```diff
@@ -928,15 +929,39 @@ public void AnalyzeCommandBase_GetFileNameFromUriWorks()
             var testCases = new Tuple<string, string>[]
             {
                 new Tuple<string, string>(null, null),
-                new Tuple<string, string>("file.txt", "file.txt"),
-                new Tuple<string, string>(@".\file.txt", "file.txt"),
-                new Tuple<string, string>(@"c:\directory\file.txt", "file.txt"),
-                new Tuple<string, string>(@"\\computer\computer\file.txt", "file.txt"),
-                new Tuple<string, string>("file://directory/file.txt", "file.txt"),
-                new Tuple<string, string>("/file.txt", "file.txt"),
-                new Tuple<string, string>("directory/file.txt", "file.txt"),
+                new Tuple<string, string>(@"file.txt", "file.txt"),
+                new Tuple<string, string>(@"file://directory/file.txt", "file.txt"),
+                new Tuple<string, string>(@"/file.txt", "file.txt"),
+                new Tuple<string, string>(@"directory/file.txt", "file.txt"),
+
+                new Tuple<string, string>(@"scheme://servername.example.com/folder/file.txt", "file.txt"),
+                new Tuple<string, string>(@"scheme://username@servername.example.com/folder/file.txt", "file.txt"),
+                new Tuple<string, string>(@"ssh://username@servername.example.com/folder/file.txt", "file.txt"),
+                new Tuple<string, string>(@"ftp://username@ftp.example.com/folder/file.txt", "file.txt"),
+                new Tuple<string, string>(@"ftp://ftp.example.com/folder/file.txt", "file.txt"),
+                new Tuple<string, string>(@"smb://servername/Share/folder/file.txt", "file.txt"),
+                new Tuple<string, string>(@"dav://example.hostname.com/folder/file.txt", "file.txt"),
+                new Tuple<string, string>(@"nfs://servername/folder/file.txt", "file.txt"),
             };
 
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                testCases.Append(new Tuple<string, string>(@".\file.txt", "file.txt"));
+                testCases.Append(new Tuple<string, string>(@"..\file.txt", "file.txt"));
+                testCases.Append(new Tuple<string, string>(@"..\directory\file.txt", "file.txt"));
+                testCases.Append(new Tuple<string, string>(@".\..\directory\file.txt", "file.txt"));
+                testCases.Append(new Tuple<string, string>(@"c:\directory\file.txt", "file.txt"));
+                testCases.Append(new Tuple<string, string>(@"\\computer\computer\file.txt", "file.txt"));
```

Comment:
>computer [](http://example.com/codeflow?start=63&length=8)

Consider UNC paths in that other test? Look at comments in that other test case area for applicability here.

Reply:
consolidated.